Index

ttimbers opened the pull request:
Swapped out US data set for {canlang} data set. I think I looked for R 4.0 changes, but nothing really came up. I still have to replace the video and screenshot of Jupyter with that of one from Jupyter lab. I will work on this as this review progresses.

ttimbers commented:
I think I'd like to Canadian-ize (or at least other-ize) the questions in the table...

ttimbers commented:
@leem44 - in this chapter with the new data set, I demonstrate `filter` to remove a couple observations where we have 0 values (i.e., no counts of that language for a given variable). Not sure how "good" of an example this is. Another option would be to `filter` out the official languages, since there are so many more counts of them than the other languages, and then perhaps I do not have to do the log scale transform (I'd need to plot that to be sure)? Thoughts?

leem44 commented on index.Rmd:
```diff
@@ -153,126 +157,121 @@ the assignment symbol is preferred and is what we will use in this course. When
-
+1. `category`: Higher level language category (describing whether the language is an Official Canadian language, an Aboriginal language, or a Non-Official and Non-Aboriginal language).
+2. `language`: Language queried about on the Canadian Census.
+3. `mother_tongue`: Total Count of Canadians from the Census who reported the language as their mother tongue. Mother tongue is generally defined as the language someone was exposed to since birth.
```
```suggestion
3. `mother_tongue`: Total count of Canadians from the Census who reported the language as their mother tongue. Mother tongue is generally defined as the language someone was exposed to since birth.
```

leem44 commented on index.Rmd:
```diff
@@ -153,126 +157,121 @@ the assignment symbol is preferred and is what we will use in this course. When
+1. `category`: Higher level language category (describing whether the language is an Official Canadian language, an Aboriginal language, or a Non-Official and Non-Aboriginal language).
+2. `language`: Language queried about on the Canadian Census.
+3. `mother_tongue`: Total Count of Canadians from the Census who reported the language as their mother tongue. Mother tongue is generally defined as the language someone was exposed to since birth.
+4. `most_at_home`: Total Count of Canadians from the Census who reported the language as spoken most often at home.
```
```suggestion
4. `most_at_home`: Total count of Canadians from the Census who reported the language as spoken most often at home.
```

leem44 commented on index.Rmd:
```diff
@@ -153,126 +157,121 @@ the assignment symbol is preferred and is what we will use in this course. When
+2. `language`: Language queried about on the Canadian Census.
+3. `mother_tongue`: Total Count of Canadians from the Census who reported the language as their mother tongue. Mother tongue is generally defined as the language someone was exposed to since birth.
+4. `most_at_home`: Total Count of Canadians from the Census who reported the language as spoken most often at home.
+5. `most_at_work`: Total Count of Canadians from the Census who reported the language as used most often at work for the population.
```
```suggestion
5. `most_at_work`: Total count of Canadians from the Census who reported the language as used most often at work for the population.
```

leem44 commented on index.Rmd:
```diff
@@ -153,126 +157,121 @@ the assignment symbol is preferred and is what we will use in this course. When
+3. `mother_tongue`: Total Count of Canadians from the Census who reported the language as their mother tongue. Mother tongue is generally defined as the language someone was exposed to since birth.
+4. `most_at_home`: Total Count of Canadians from the Census who reported the language as spoken most often at home.
+5. `most_at_work`: Total Count of Canadians from the Census who reported the language as used most often at work for the population.
+6. `lang_known`: Total Count of Canadians from the Census who reported knowledge of language for the population in private households.
```
```suggestion
6. `lang_known`: Total count of Canadians from the Census who reported knowledge of the language for the population in private households.
```

leem44 commented on index.Rmd:
```diff
@@ -304,101 +303,116 @@ ggplot(us_data, aes(x = med_income, y = med_prop_val)) +
 </center>
 
-From this visualization we see that for the 52 US regions in this data set, as median household income increases so does median property value. When we see 
+Most of the data points from the 214 observations in this data set are bunched up in the lower left-handside of this visualization. This is because many many more people in Canada speak the two official languages (English and French). Thus to answer our question, we will need to adjust the scale of the x and y axes so that they are on a log scale. We can again add additional layers to the plot ob ...
```
```suggestion
Most of the data points from the 214 observations in this data set are bunched up in the lower-left hand side of this visualization. This is because many more people in Canada speak the two official languages (English and French). Thus to answer our question, we will need to adjust the scale of the x and y axes so that they are on a log scale. We can again add additional layers to the plot object using the `+` symbol to do this:
```

leem44 commented on index.Rmd:
```diff
@@ -304,101 +303,116 @@ ggplot(us_data, aes(x = med_income, y = med_prop_val)) +
 ### Coloring points by group
-Now we'll move onto the second part of our exploratory data analysis question: when considering the relationship between median household income and median property 
-value, is there a pattern in which party each state voted for in the 2016 US election? One common way to explore this is to colour the data points on the 
+Now we'll move onto the second part of our exploratory data analysis question: when considering the relationship between the number of people who have a language as their mother tongue and the number of people who speak that language at home, is there a pattern in the strength of this relationship in the higher level language categories (Official languages, Aboriginal languages, or non-official a ...
```
```suggestion
Now we'll move onto the second part of our exploratory data analysis question: when considering the relationship between the number of people who have a language as their mother tongue and the number of people who speak that language at home, is there a pattern in the strength of this relationship in the higher-level language categories (Official languages, Aboriginal languages, or non-official and non-Aboriginal languages)? One common way to explore this is to colour the data points on the 
```

leem44 commented on index.Rmd:
```diff
@@ -153,126 +157,121 @@ the assignment symbol is preferred and is what we will use in this course. When
 
 ## Exploring data with visualizations
 
 Creating effective data visualizations is an essential piece to any data analysis. For the remainder of Chapter 1, we will learn how to use
```
Suggestion for this paragraph: 

Creating effective data visualizations is an essential piece of any data analysis. For the remainder of Chapter 1, we will learn how to use functions from the tidyverse package to make visualizations that let us explore relationships in data. In particular, we’ll develop a visualization of the language data collected in the 2016 Canadian census we’ve been working with that will help us understand two potential relationships in the data. First, we will explore the relationship between the number of people who speak a language as their mother tongue and the number of people who speak it as their primary language at home. And second, we will explore whether there a pattern in the strength of this relationship in the higher-level language categories (Official languages, Aboriginal languages, or non-official and non-Aboriginal languages). These questions are examples of exploratory data analysis questions: we are looking for relationships and patterns within the data set we have, but are not trying to generalize what we find to others beyond this data set.

leem44 commented on index.Rmd:
```diff
@@ -304,101 +303,116 @@ ggplot(us_data, aes(x = med_income, y = med_prop_val)) +
 scatter plot we have already created by group/category. For example,
-given that we have the party each state voted for in the 2016 US Presidential election in the column named `party`, we can colour the points in our previous 
-scatter plot to represent who each stated voted for. 
+given that we have the higher level language category for each language recorded in the 2016 Canadian census, we can colour the points in our previous 
```
```suggestion
given that we have the higher-level language category for each language recorded in the 2016 Canadian census, we can colour the points in our previous 
```

leem44 commented on index.Rmd:
```diff
@@ -304,101 +303,116 @@ ggplot(us_data, aes(x = med_income, y = med_prop_val)) +
 which focuses on data visualization. Then, we will explicitly teach you how to do this yourself, and how to not over-state or over-interpret the results 
 from a visualization.
 
 ### Coloring points by group
```
```suggestion
### Colouring points by group
```

leem44 commented on index.Rmd:
```diff
@@ -153,126 +157,121 @@ the assignment symbol is preferred and is what we will use in this course. When
+first, the relationship between the number of people who speak a language as their mother tongue and the number of people who speak that language as their primary spoken language at home, and second, whether there is a pattern in the strength of this relationship in the higher level language categories (Official languages, Aboriginal languages, or non-official and non-Aboriginal languages). This  ...
 question: we are looking for relationships and patterns within the data set we have, but are not trying to generalize what we find beyond this data set. 
 
 ### Using `ggplot` to create a scatter plot
```
The code for the plots in this section are centred so it looks a little off

leem44 commented on index.Rmd:
```diff
@@ -304,101 +303,116 @@ ggplot(us_data, aes(x = med_income, y = med_prop_val)) +
-given that we have the party each state voted for in the 2016 US Presidential election in the column named `party`, we can colour the points in our previous 
-scatter plot to represent who each stated voted for. 
+given that we have the higher level language category for each language recorded in the 2016 Canadian census, we can colour the points in our previous 
+scatter plot to represent the higher level language category each language belongs to. 
```
```suggestion
scatter plot to represent each language's higher-level language category.
```

leem44 commented on index.Rmd:
```diff
@@ -304,101 +303,116 @@ ggplot(us_data, aes(x = med_income, y = med_prop_val)) +
 
-That explains it! That row in the dataset is actually not a US state, but rather the US territory of Peurto Rico. Similar to other US territories, residents of Puerto Rico 
-cannot vote in presidential elections. Hence the "not applicable" label. Let's remove this row from the data frame and rename the data frame `vote_data`. 
+Let's remove these rows from the data frame`. 
```
```suggestion
Let's remove these rows from the data frame. 
```

leem44 commented on index.Rmd:
```diff
@@ -304,101 +303,116 @@ ggplot(us_data, aes(x = med_income, y = med_prop_val)) +
-Most of the US States who voted for the Republican candidate in the 2016 US Presidential election had lower median household income and lower median property values (data points primarily 
-fall in lower left-hand side of the scatter plot), whereas most of the US states who voted for the Democratic candidate in the 2016 US Presidential election had higher median household 
-income and higher median property values (data points primarily fall in the upper right-hand side of the scatter plot).
+What do we see when considering the second part of our exploratory question? Do we see a pattern in the relationship between the number of people who speak a language as their mother tongue and the number of people who speak a language as their primary spoken language at home for the different higher-level language categories? Probably not! 
```
```suggestion
What do we see when considering the second part of our exploratory question? Do we see a difference in the pattern of the relationship between the number of people who speak a language as their mother tongue and the number of people who speak a language as their primary spoken language at home between higher-level language categories? Probably not!
```

leem44 commented on index.Rmd:
```diff
@@ -304,101 +303,116 @@ ggplot(us_data, aes(x = med_income, y = med_prop_val)) +
+
+For each higher-level language category there appears to be a positive relationship between the number of people who speak a language as their mother tongue and the number of people who speak a language as their primary spoken language at home. This relationship looks similar, regardless of the category. 
+
+Does this mean that the relationship between the number of people who speak a language as their mother tongue and the number of people who speak a language as their primary spoken language at home is positive for all languages in the world? Or could we use this data visualization on its own to predict how many people have a given language as their mother tongue if we know how many people speak it ...
```
```suggestion
Does this mean that this relationship is positive for all languages in the world? Can we use this data visualization on its own to predict how many people have a given language as their mother tongue if we know how many people speak it as their primary language at home?
```

leem44 commented on index.Rmd:
```diff
@@ -304,101 +303,116 @@ ggplot(us_data, aes(x = med_income, y = med_prop_val)) +
 
-Does this mean that rich states usually vote for Democrats and poorer states generally vote for Republicans? Or could we use this data visualization on its own to predict which party each 
-state will vote for in the next presidential election? The answer to both these questions is "no." What we can do with this exploratory data analysis is create new hypotheses, ideas, 
+The answer to both these questions is "no." What we can do with this exploratory data analysis is create new hypotheses, ideas, 
```
```suggestion
The answer to both these questions is "no." However, with this exploratory data analysis, we can create new hypotheses, ideas,
```

leem44 commented:
@ttimbers The chapter looks really good! I had some comments that were mostly quite minor. 
- Yes I agree with Canadian-izing the table (or at least removing so much of the American context)
- I like the idea of removing the official languages if it helps with the plot. The log transformation might be too much for some students?

ttimbers commented on index.Rmd:
````diff
@@ -153,126 +157,121 @@ the assignment symbol is preferred and is what we will use in this course. When
+can_lang
 ```
 
 ### Using `ggplot` to create a scatter plot
````
fixed in e20b1f8bb7

ttimbers commented on index.Rmd:
```diff
@@ -153,126 +157,121 @@ the assignment symbol is preferred and is what we will use in this course. When
-Taking another look at our dataset below, we can immediately see that the three columns (or variables) we are interested in visualizing---median household 
-income, median property value, and election result---are all in separate columns. In addition, there is a single row (or observation) for each state. 
+Taking another look at our data set below, we can immediately see that the three columns (or variables) we are interested in visualizing---mother tongue, language spoken most at home, and higher level language category---are all in separate columns. In addition, there is a single row (or observation) for each language. 
 The data are therefore in what we call a *tidy data* format. 
```
Added that we will formally introduce tidy data in chapter 3 in 1e71bcff9964f

ttimbers replied:
Added that we will formally introduce tidy data in chapter 3 in 1e71bcff9964f

ttimbers commented:
> I like the idea of removing the official languages if it helps with the plot. I do wonder if the log transformation might be too much for some students?

Sigh, it doesn't help that much... 

<img width="853" alt="Screen Shot 2020-12-08 at 10 42 39 PM" src="https://user-images.githubusercontent.com/6865016/101594553-bfca6980-39a6-11eb-8792-6641de5fee82.png">

So removing the 0's is my only idea right now... Alternatively we could select only the columns we want to work with? That might be better than filtering out the 0's?

ttimbers commented on index.Rmd:
```diff
@@ -304,101 +303,116 @@ ggplot(us_data, aes(x = med_income, y = med_prop_val)) +
 two variables do this, we call this a *positive relationship*. Because the increasing pattern is fairly clear 
-(not fuzzy) we can say that the relationship is strong. Because of the data point in the lower left-hand corner, drawing a straight line through these 
-points wouldn't fit very well. When a straight-line doesn't fit the data well we say that it's non-linear. However, we should have caution when using one point to claim non-linearity. As we will see later, this might be due to a single point not really belonging in the data set (this is often called an outlier).
+(not fuzzy) we can say that the relationship is strong. Because drawing a straight line through these 
```
Changed language to that points being close together in d888bfcc9be09

ttimbers commented on index.Rmd:
````diff
@@ -153,126 +157,121 @@ the assignment symbol is preferred and is what we will use in this course. When
+can_lang <- read_csv("data/can_lang.csv")
 ```
 
 Wait a minute! Nothing happened this time! Or at least it looks like that. But actually something did happen: the data was read in and now has the name `us_data` associated with it. 
````
Thanks for catching this!

ttimbers commented:
OK, I addressed all your comments @leem44 , but what is still left to tidy up:

1. Should we select relevant columns instead of remove 0's?

2. I need to update the screen shot and video

Would like your feedback still on the Canadianized Q's.

ttimbers commented:
OK, from our zoom conversation yesterday, instead of selecting relevant columns, or removing 0's, we will use mutate to make the counts into percentages as that is more realistic.

ttimbers commented:
@leem44 & @trevorcampbell - I have now changed the wrangling in Chapter 1 (index) to use `mutate` to plot proportions instead of counts. There are a couple paragraphs of new narration to accompany this. Would appreciate a review before we merge this.

leem44 commented on index.Rmd:
```diff
@@ -53,12 +57,12 @@ For each of these problems, we map them to the type of data analysis question be
-| Mechanistic | A question that asks about the underlying mechanism of the observed patterns, trends, or relationship (i.e., how does it happen?) | How does wealth lead to voting for a certain political party candidate in the US Presidential election? |
+| Descriptive | A question which asks about summarized characteristics of a data set without interpretation (i.e., report a fact). | How many people live in each province or territory in Canada? |
+| Exploratory | A question asks if there are patterns, trends, or relationships within a single data set. Often used to propose hypotheses for future study. | Does political party voting change with indicators of wealth in a set of data collected from groups of individuals from several regions in Canada? |
+| Inferential | A question that looks for patterns, trends, or relationships in a single data set **and** also asks for quantification of how applicable these findings are to the wider population. | Does political party voting change with indicators of wealth in Canada? |
```
For the questions in the table, maybe we can differentiate the "Inferential" a little more by indicating that this question is referring to a population. And that way we can contrast an exploratory and inferential question by making it essentially the same question but changing the group it's referring to, something like: "Does political party voting change with indicators of wealth for every single person in Canada?" (or something along those lines) and then exploratory is the same question but just about the data

ttimbers replied:
Fixed in 6fdf3cf

leem44 commented on index.Rmd:
```diff
@@ -304,101 +303,127 @@ ggplot(us_data, aes(x = med_income, y = med_prop_val)) +
 
 One common and easy way to format your `ggplot` visualization is to add additional layers to the plot object using the `+` symbol. 
```
Maybe here we can say something commenting on the plot we made above in 6.1.2 just to give a little more motivation for the code that follows. Kind of like: Yes, we have plotted a scatterplot, but we aren't done yet -- there's more that we can do to improve the interpretability of the plot. For instance, most people won't know what "mother_tongue" means without more explanation etc...

ttimbers replied:
Addressed in 36c7989

leem44 commented on index.Rmd:
````diff
@@ -304,101 +303,127 @@ ggplot(us_data, aes(x = med_income, y = med_prop_val)) +
 
-```{r relevel factors for colour, include=FALSE}
-us_data <- us_data %>% mutate(party = factor(party, levels = c("Republican", "Not Applicable", "Democratic")))
+```{r}
````
I think this code block needs a little explanation before it

ttimbers replied:
I am not sure what this comment is referring to, can you please clarify @leem44 ?

leem44 commented on index.Rmd:
````diff
@@ -304,101 +303,127 @@ ggplot(us_data, aes(x = med_income, y = med_prop_val)) +
-  ylab("Median property value (USD)")
-```
-</center>
+What does it mean that `r format(english_mother_tongue, scientific = FALSE, big.mark = ",")` people reported that their mother tongue was English in the 2016 Canadian census? To really understand this number, we need context, in particular, how many people are there were in Canada when this data was collected? From the 2016 Canadian census profile we can see that the population was reported to be ...
````
```suggestion
What does it mean that `r format(english_mother_tongue, scientific = FALSE, big.mark = ",")` people reported that their mother tongue was English in the 2016 Canadian census? To really understand this number, we need context. In particular, how many people were in Canada when this data was collected? From the 2016 Canadian census profile, we can see that the population was reported to be `r format(census_popn, scientific = FALSE, big.mark = ",")` people. The absolute count of the number of people who report that English is their mother tongue is much more meaningful when we report it in this context. We can even go a step further and transform this absolute count to a relative frequency, or proportion, so that we can represent this as a single meaningful number in our data visualizations. We can do this by dividing the number of people reporting a given language as their mother tongue by the number of people who live in Canada. For example, the proportion of people who reported that their mother tongue was English in the 2016 Canadian census was `r format(round(english_mother_tongue/census_popn, 2), scientific = FALSE, big.mark = ",")`.
```

ttimbers replied:
Accepted this, but will modify further as I have no idea why I thought I should call it an absolute count as opposed to a count.

leem44 commented on index.Rmd:
````diff
@@ -304,101 +303,127 @@ ggplot(us_data, aes(x = med_income, y = med_prop_val)) +
 ```
+</center>
+
+From the visualization above, we can now clearly see that not just a lot, but that the majority Canadians reported English as their mother tongue and as the language they speak most often at home. Changing the units to include this context increases our understanding and allows us to better interpret the numbers in our data set.
````
> 
```suggestion
From the visualization above, we can now clearly see that not just a lot, but that the majority of Canadians reported English as their mother tongue and as the language they speak most often at home. Changing the units to include this context increases our understanding and allows us to interpret the numbers in our data set better.
```

leem44 commented on index.Rmd:
```diff
@@ -304,101 +303,127 @@ ggplot(us_data, aes(x = med_income, y = med_prop_val)) +
+From the visualization above, we can now clearly see that not just a lot, but that the majority Canadians reported English as their mother tongue and as the language they speak most often at home. Changing the units to include this context increases our understanding and allows us to better interpret the numbers in our data set.
+
+### Coloring points by group
+Now we'll move onto the second part of our exploratory data analysis question: when considering the relationship between the number of people who have a language as their mother tongue and the number of people who speak that language at home, is there a pattern in the strength of this relationship in the higher level language categories (Official languages, Aboriginal languages, or non-official a ...
```
```suggestion
Now we'll move onto the second part of our exploratory data analysis question: when considering the relationship between the number of people who have a language as their mother tongue and the number of people who speak that language at home, is there a pattern in the strength of this relationship in the higher-level language categories (Official languages, Aboriginal languages, or non-official and non-Aboriginal languages)? One common way to explore this is to colour the data points on the 
```

leem44 commented on index.Rmd:
```diff
@@ -304,101 +303,127 @@ ggplot(us_data, aes(x = med_income, y = med_prop_val)) +
-Most of the US States who voted for the Republican candidate in the 2016 US Presidential election had lower median household income and lower median property values (data points primarily 
-fall in lower left-hand side of the scatter plot), whereas most of the US states who voted for the Democratic candidate in the 2016 US Presidential election had higher median household 
-income and higher median property values (data points primarily fall in the upper right-hand side of the scatter plot).
+To do this we modify our scatter plot code above. Specifically, we will add an argument to the `aes` function, specifying that the points should be coloured by the `category` column:
```
```suggestion
To do this, we modify our scatter plot code above. Specifically, we will add an argument to the `aes` function, specifying that the points should be coloured by the `category` column:
```

leem44 commented on index.Rmd:
```diff
@@ -304,101 +303,127 @@ ggplot(us_data, aes(x = med_income, y = med_prop_val)) +
+                     color = category,
+                     shape = category)) +
+  geom_point(alpha = 0.7) +
+  scale_color_manual(values = c("blue3","red3","black")) +
```
For this last plot, I would colour it so it's the same as the one above since were "putting it all together" it'll show that its the same code as above a bit more

ttimbers replied:
OK, on Slack we discussed that I will keep the more effective plot, but in the writing above state that I add extra layers to do things, and then inline comment the extra new stuff. @leem44  agreed. This is reflected in 
b516099

leem44 commented:
@ttimbers Here are my comments -- just some minor things! Looks good!

ttimbers commented:
Thanks for the comments and feedback Melissa! I will loop back to these as soon as I have finished a draft of the version controll chapter!

ttimbers commented:
All addressed so merging 🚢 ! Thanks @leem44 for the thorough review!! 🙏
